addons/mail/static/src/components/file_uploader/file_uploader.js

```diff
@@ -101,7 +101,10 @@ class FileUploader extends Component {
                 attachment.isTemporary &&
                 attachment.filename === file.name
             );
-
+            if (!uploadingAttachment) {
+                // The current attachment may have been removed while awaiting for the previous fetch.
```

Comment:
Took me some time to figure out what the comment means. I had to read the whole function and make mental effort to understand how it's functionally possible to lose an attachment being uploaded in model.

All this to say comment is not good. I suggest this comment, which is much nicer IMO:
```js
// Uploading attachment no longer exists.
// This happens when an uploading attachment is being deleted by user.
```

I would not put too much details (such as it's being deleted when a previous uploading attachment is being uploaded), because this is too intimidating to be put in small comment, and it's heavily dependent on implementation detail of this function, and it's best just read the function for that matter.